ui/src/bootstrap.rs

```diff
@@ -59,18 +62,24 @@ pub(crate) fn use_boostrap<'a>(
 
 pub fn set_app_panic_hook() {
     panic::set_hook(Box::new(|panic_info| {
-        let intro = match panic_info.payload().downcast_ref::<&str>() {
-            Some(s) => format!("panic occurred: {s:?}"),
-            None => "panic occurred".into(),
-        };
-        let location = match panic_info.location() {
-            Some(loc) => format!(" at file {}, line {}", loc.file(), loc.line()),
-            None => "".into(),
-        };
-
         let logs = logger::dump_logs();
-        let crash_report = format!("{intro}{location}\n{logs}\n");
+        let crash_report = format!("{panic_info}\n{logs}\n");
+        let backtrace = Backtrace::force_capture();
+        let time = Local::now().format("%d-%m-%Y_%H-%M-%S").to_string();
+        let path = STATIC_ARGS.crash_logs.join(format!("{}.log", time));
+        let log = format!(
+            "Uplink crashed:\n{}\nBackTrace:\n{}",
+            crash_report, backtrace
+        );
         println!("{crash_report}");
+        if let Some(p) = path.parent() {
+            if fs::create_dir_all(p).is_ok() {
```

Comment:
It might be best to do `let _ = fs::create_dir_all(p);` and ignore any errors since `fs::write` would give the reason for why it failed to write in that path. If we want to still check to make sure it does create the directories, then we should catch the error instead by doing something along the lines of 

```rust
if let Err(e) = fs::create_dir_all(p) {
   println!("Unable to create directory: {e}");
   return;
}

match fs::write(&path, log) {
    Ok(_) => println!("The crash has been saved to {}", path.to_string_lossy()),
    Err(e) => println!("Unable to save crash log {}", e),
}
```